File: api/src/main/java/com/hearlers/gateway/application/user/UserService.java
package com.hearlers.gateway.application.user;

import com.hearlers.api.proto.v1.model.User;
import com.hearlers.api.proto.v1.service.FindUserByUserIdRequest;
import com.hearlers.api.proto.v1.service.UpdateUserRequest;

public interface UserService {
    User updateUser(UpdateUserRequest request);
    User findUserByUserId(FindUserByUserIdRequest request);
}

File: usecase/user-usecase/src/main/kotlin/com/hearlers/gateway/UserService.kt
package com.hearlers.gateway

import com.hearlers.api.proto.v1.model.User
import com.hearlers.api.proto.v1.model.UserTracking
import com.hearlers.api.proto.v1.service.FindUserByUserIdRequest
import com.hearlers.api.proto.v1.service.UpdateTrackingRequest
import com.hearlers.api.proto.v1.service.UpdateUserRequest
import com.hearlers.gateway.port.UserPort
import org.springframework.stereotype.Service

@Service
class UserService(
    private val userPort: UserPort
) : UserUseCase {
    override fun updateUser(request: UpdateUserRequest): User {
        return userPort.updateUser(request)
    }

    override fun findUserByUserId(request: FindUserByUserIdRequest): User? {
        return userPort.findUserByUserId(request)
    }

    override fun updateUserTracking(request: UpdateTrackingRequest): UserTracking {
        return userPort.upsertUserTracking(request)
    }

    override fun getUserTrackingByUserId(userId: String): UserTracking {
        return userPort.getUserTrackingByUserId(userId)
    }
}